### PR TITLE
[GLUTEN-1534] [VL] reduce pre-projection for hash join

### PR DESCRIPTION
**Pull request by zhli1142015**

### CODE REVIEW

File: gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala [hunk 1]
```diff
@@ -147,8 +147,11 @@ trait HashJoinLikeExecTransformer
         .zip(rightKeys.map(_.dataType))
         .forall(types => sameType(types._1, types._2)),
       "Join keys from two sides should have same length and types")
-    val lkeys = HashJoin.rewriteKeyExpr(leftKeys)
-    val rkeys = HashJoin.rewriteKeyExpr(rightKeys)
+    // Spark has an improvement which would patch integer joins keys to a Long value.
+    // But this improvement would cause extra projet before hash join in velox, disalbe
```
Comment (zhztheplayer) [inline gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala:151]: typo: disalbe -> disabling

Comment (zhli1142015) [reply gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala]: Thanks.

**Comment by FelixYBW** [comment]
Good work. Jenkins TPCDS Performance is improved by 5%, 80 of 103 queries have better performance.

**Review by zhztheplayer — APPROVED** [review]
Thanks @zhli1142015 !

**Review by FelixYBW — APPROVED** [review]

**Comment by zhli1142015** [comment]
Thanks @FelixYBW  and @zhztheplayer  for reviewing.

**Pull request merged**